lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -419,6 +421,41 @@ def add_stored_credentials(post, options = {})
         post[:payment_method_options][:card][:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
       end
 
+      def add_stored_credential_transaction_type(post, options = {})
+        stored_credential = options[:stored_credential]
+        # Do not add anything unless these are present.
+        return unless stored_credential[:reason_type] && stored_credential[:initiator]
+
+        # Not compatible with off_session parameter.
+        options.delete(:off_session)
+
+        if stored_credential[:initial_transaction]
+          # Initial transactions must by CIT
+          return unless stored_credential[:initiator] == 'cardholder'
+
+          if stored_credential[:reason_type] == 'unscheduled'
```

Comment:
The `if/else` nesting is a bit hard to read. Maybe try handling the nested `if/else` conditional in a separate method?